DRY out application instructions

fureigh opened the pull request:
Direct all applicants to the 'How to Apply' page so any important information there, including instructions related to #54, reaches them.

@arowla, sending this to you for review. There must be a more robust way to construct this link.

fureigh committed:
DRY out application instructions

Direct all applicants to the 'How to Apply' page so any important information there, including instructions related to #54, reaches them.

arowla commented on pages/interview-process.md:
```diff
@@ -6,7 +6,7 @@ title: Interview process
 ### Phone screen
 
-Apply to 18F by emailing your resume to [join18f@gsa.gov](mailto:join18f@gsa.gov). Within three weeks our team will review your resume, and if you seem like a good fit, we'll reach out to schedule a 30-minute phone call. Plan to talk about your skills and experience, what you're passionate about, and the work we do here. 
+[Apply to 18F](/how-to-apply). Within three weeks our team will review your resume, and if you seem like a good fit, we'll reach out to schedule a 30-minute phone call. Plan to talk about your skills and experience, what you're passionate about, and the work we do here. 
```
According to [this](https://pages.18f.gov/guides-template/add-a-new-page/), the link needs to be prefixed with `{{ site.baseurl }}`.

fureigh committed:
Add site.baseurl to Apply page link

Per @arowla's feedback and https://pages.18f.gov/guides-template/add-a-new-page/.

fureigh commented:
@arowla Thanks for pointing out the docs! This is now ready for another look.
